navzam/communication-services-web-calling-hero
# Improve management of user/group/ACS info

## Purpose

* Store all user/group/ACS info in existing Table Storage.
* Store appointment info separately from "appointment user" info. This lets us store group-wide details like the ACS chat thread ID. Previously we only stored which users were part of which group.
* Get chat thread ID from Storage when needed, such as when sending the "file event" into the chat thread.
* Collect user name on home screen instead of configuration screen. This is needed because the configuration screen uses the user ID to initialize the `CallAgent`, so we need the user name before that. The home screen can be thought of as the login step in a real authenticated app.

## Does this introduce a breaking change?

```
[ ] Yes
[ x ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?


```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
* Start a group call with multiple users.
* Send chat messages and files in the group.
* Observe Storage tables to see the data stored.
